lib/src/backend.rs

```diff
@@ -147,7 +148,7 @@ content_hash! {
     }
 }
 
-pub type SigningFn = Box<dyn FnMut(&[u8]) -> BackendResult<Vec<u8>>>;
+pub type SigningFn = Box<dyn FnMut(&[u8]) -> Result<Vec<u8>, SignError>>;
```

Comment:
nit: unrelated to this change, I think `SigningFn` could be passed around as `Option<&mut dyn FnMut(...>` (or `Option<&dyn Fn(...>`.)

```rust
type SigningFn<'a> = dyn FnMut... + 'a;

fn write_commit(
    ...,
    sign_with: Option<&mut SigningFn>,
)
```

It will help if we want to pass `&Signer` to `write_commit()` without cloning the owner.

I can send a follow up patch if you think that's good idea.

Reply:
Sure, sounds great